File: 0926-flip-string-to-monotone-increasing/0926-flip-string-to-monotone-increasing.java
class Solution {
    public int minFlipsMonoIncr(String s) {
        int needToConvertToZero = 0, needToConvertToOne = 0;
        for(int i = 0 ; i < s.length() ; i++)
        {
            int cur = s.charAt(i) - '0';

            needToConvertToOne = Math.min(needToConvertToOne, needToConvertToZero) + (cur == 0 ? 1 : 0);
            needToConvertToZero += cur;
        }
        return Math.min(needToConvertToZero, needToConvertToOne);
    }
}

File: 0926-flip-string-to-monotone-increasing/0926-flip-string-to-monotone-increasing.cpp
class Solution {
public:
    int minFlipsMonoIncr(string s) {
        int sum = 0, cur = 0;
        for(auto i : s)
            sum += (i - '0');

        int len = s.length();
        int ans = len;
        for(int i = 0 ; i <= len ; i++)
        {
            int rem = sum - cur;
            int curAns = cur;
            curAns += (len - i - rem);
            
            ans = min(ans, curAns);
            if(i == len)
                break;
            cur += (s[i] - '0');
        }
        return ans;
    }
};